### DISCUSSION TITLE
Installation is successful but Rosetta is not running!

### DISCUSSION THREAD

**Discussion by intomybioverse** [RosettaCommons/rosetta#73] [Build & Install] [answered]
I have installed rosetta in my linux (Ubuntu 20.04) system by following the installation guide mentioned in the website. 
https://docs.rosettacommons.org/docs/latest/getting_started/Getting-Started#:~:text=development%20of%20Rosetta.)-,Local%20installation%20and%20use%20of%20Rosetta,comes%20as%20a%20tar%20archive.

However, the program is not running with the command rosettaDNA.static.linuxgccrelease @arg. Even the test for compilation did not run because of the FileNotFoundError. Although, g++ is present in user/bin/g++, the compiler compatibility test shows g++ is not found. 
Please guide me on this? How to I proceed after the installation process? How do I make the program work?

**Comment by dgront** [comment]
Could you paste the error message you got after running Rosetta? This would be helpful. Overall, I see three possible reasons:
1) `rosettaDNA.static.linuxgccrelease` is not there, kindly check the file. Please also note this is a symbolic link, so check also if the original file the link is pointing to can be located
2) are you using a file with command line flags with your `@arg`? Is the `arg` an actual file in the current folder?
3) do you mention some input files in your `arg`? Double check if they actually exist

**Comment by intomybioverse** [comment]
Here is the error message.

soumya@soumya-CELSIUS-R970:~/Downloads/rosetta.source.release-371/main/tests/integration/tests/dna_interface_design$ rosettaDNA.static.linuxgccrelease @
design.flags
rosettaDNA.static.linuxgccrelease: command not found
design.flags: command not found

I am trying to change the specificity of a protein so that it binds to a desired DNA sequence. For this, I am following the protocol mentioned below. 

Thyme S, Baker D. Redesigning the specificity of protein-DNA interactions with Rosetta. Methods Mol Biol. 2014;1123:265-82. doi: 10.1007/978-1-62703-968-0_17. PMID: 24510272.

I am trying to run a demo test first as given in the rosettaDNA applications on the Rosetta website (https://docs.rosettacommons.org/docs/latest/application_documentation/design/rosetta-dna)
source directory: tests/integration/tests/dna_interface_design/

Even a terminal opened through the rosetta_scrpits/RosettaDNA directory, I am getting the command not found error. 
(base) soumya@soumya-CELSIUS-R970:~/Downloads/rosetta.source.release-371/main/demos/protocol_capture/rosetta_scripts/RosettaDNA$ rosettaDNA.static.linuxgccrelease @design.flags
rosettaDNA.static.linuxgccrelease: command not found

Yes, the design.flags file is in the same directory.
Here are the contents,
-in:ignore_unrecognized_res
-file:s msoleft.pdb
-score:weights dna
-use_input_sc
-run:output_hbond_info
-sparse_pdb_output
-score:output_residue_energies
-run:min_type dfpmin_armijo
-run:min_tolerance 0.0001
-adducts dna_major_groove_water
-out:levels all:warning protocols.dna:info protocols.multistate_design:info protocols.genetic_algorithm:info apps:info protocols.loops:info core.io.pdb:info
-jd2:dd_parser
-parser:protocol design.script
-overwrite
-out:prefix design_
-exclude_patches patches/carbohydrates/2-amination.txt patches/carbohydrates/2-branch.txt patches/carbohydrates/3-amination.txt patches/carbohydrates/3-branch.txt patches/carbohydrates/3-methylation.txt patches/carbohydrates/5-acetylation.txt patches/carbohydrates/6-branch.txt patches/carbohydrates/branch_lower_term.txt patches/carbohydrates/cutpoint_lower.txt patches/carbohydrates/cutpoint_upper.txt patches/carbohydrates/lower_terminus.txt patches/carbohydrates/Me_glycoside.txt patches/carbohydrates/N-acetyl-2-amination.txt patches/carbohydrates/N-linked_glycosylation.txt patches/carbohydrates/upper_terminus.txt patches/branching/aryl-C-conjugated.txt patches/branching/aryl-O-conjugated.txt patches/branching/C-terminal_conjugation.txt patches/branching/N-linked_conjugation.txt patches/branching/O-linked_conjugation.txt patches/branching/phg_cd1_conjugation.txt patches/branching/phg_cd2_conjugation.txt patches/branching/S-linked_conjugation.txt patches/branching/sidechain_carboxyl_conjugation.txt patches/branching/sidechain_electrophile_conjugation.txt patches/NtermConnect.txt patches/CtermConnect.txt patches/oop_post.txt patches/oop_pre.txt patches/hbs_post.txt patches/hbs_pre.txt patches/a3b_hbs_post.txt patches/a3b_hbs_pre.txt 

Here is the information on the available gcc.
(base) soumya@soumya-CELSIUS-R970:~$ which gcc
/usr/bin/gcc
(base) soumya@soumya-CELSIUS-R970:~$ gcc --version
gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

How do I run the program? Please help out.

**Reply by ajasja** [reply]
You probably should be using LigandMPNN for the redesign (https://github.com/dauparas/LigandMPNN), the success rates will be better. Good luck.

**Comment by roccomoretti** [comment]
Generally speaking, the Rosetta commands aren't added to your general path. So you'll have to explicitly provide the full path to the command when you run.

So instead of 

```rosettaDNA.static.linuxgccrelease @design.flags```

try

```~/Downloads/rosetta.source.release-371/main/source/bin/rosettaDNA.static.linuxgccrelease @design.flags```

Depending on where/how you installed Rosetta, the path might be different. You can always do a

```ls ~/Downloads/rosetta.source.release-371/main/source/bin/rosettaDNA.*```

To see if the application is installed in that path.

Note also that depending on how Rosetta was compiled, the name of the executable may be different. It could be rosettaDNA.default.linuxgccrelease or rosettaDNA.default.macclangrelease or something else. The first part of the application name is always the same, but the second two bits can be different depending on the exact compilation details. (Generally static versus default, clang versus gcc versus icc, and linux versus mac doesn't really matter. The key differences for most users are debug versus release, and things like `tensorflow` and `mpi`, if you're using the tensorflow or mpi features.) -- The `ls` command above should give you the details about which compiled versions you have on your machine.

**Comment by intomybioverse** [comment]
Yes, I followed your instructions and here is the new error message. 

(base) soumya@soumya-CELSIUS-R970:~$ ls ~/Downloads/rosetta.source.release-371/main/source/bin/rosettaDNA.*
/home/soumya/Downloads/rosetta.source.release-371/main/source/bin/rosettaDNA.default.linuxgccrelease
/home/soumya/Downloads/rosetta.source.release-371/main/source/bin/rosettaDNA.linuxgccrelease
(base) soumya@soumya-CELSIUS-R970:~$ /home/soumya/Downloads/rosetta.source.release-371/main/source/bin/rosettaDNA.linuxgccrelease @design.flags
 

[ ERROR ]: Caught exception:


File: src/utility/options/OptionCollection.cc:572
Option file open failed for: 'design.flags'


AN INTERNAL ERROR HAS OCCURED. PLEASE SEE THE CONTENTS OF ROSETTA_CRASH.log FOR DETAILS.


Here are the contents of the ROSETTA_CRASH.log for your reference. 
##############################################################################################################

#Rosetta crash log. Please submit the contents of this file to http://crash.rosettacommons.org/ or use the script Rosetta/main/source/scripts/python/public/report_crashes.py

[START_CRASH_REPORT]
[ROSETTA_VERSION]: 2024.09+release.06b3cf8
[COMMIT_DATE]: 2024-03-07T14:59:22.624665
[APPLICATION]: /home/soumya/Downloads/rosetta.source.release-371/main/source/bin/rosettaDNA.default.linuxgccrelease
[MODE]: Release
[EXTRAS]: default
[OS]: GNU/Linux
[COMPILER]: GCC version "11.4.0"
[STDLIB]: libstdc++ version 20230528
[START_OPTIONS]


[END_OPTIONS]

[START_BACKTRACE]: RAW_LIBC
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libutility.so(backtrace_string[abi:cxx11](int)+0x5a) [0x7a63c2fab35a]
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libutility.so(utility::excn::Exception::Exception(char const*, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0xe0) [0x7a63c2fe9310]
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libutility.so(utility::UtilityExitException::UtilityExitException(char const*, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0x113) [0x7a63c2fb04d3]
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libutility.so(utility::exit(char const*, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)+0x3b) [0x7a63c2fb011b]
/home/soumya/Downloads/rosetta.source.release-371/main/source/bin/rosettaDNA.default.linuxgccrelease(+0x3e18) [0x5ee95f198e18]
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libprotocols.1.so(protocols::jd2::JobDistributor::run_one_job(std::shared_ptr<protocols::moves::Mover>&, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, unsigned long&, unsigned long&, bool)+0xe57) [0x7a63c4111377]
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libprotocols.1.so(protocols::jd2::JobDistributor::go_main(std::shared_ptr<protocols::moves::Mover>)+0x238) [0x7a63c4113388]
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libprotocols.1.so(protocols::jd2::FileSystemJobDistributor::go(std::shared_ptr<protocols::moves::Mover>)+0x59) [0x7a63c40fbf59]
/home/soumya/Downloads/rosetta.source.release-371/main/source/build/src/release/linux/6.5/64/x86/gcc/11/default/libprotocols.1.so(protocols::jd2::JobDistributor::go(std::shared_ptr<protocols::moves::Mover>, std::shared_ptr<protocols::jd2::JobOutputter>)+0x18f) [0x7a63c4113caf]
/home/soumya/Downloads/rosetta.source.release-371/main/source/bin/rosettaDNA.default.linuxgccrelease(+0x3815) [0x5ee95f198815]
/home/soumya/Downloads/rosetta.source.release-371/main/source/bin/rosettaDNA.default.linuxgccrelease(+0x34b1) [0x5ee95f1984b1]
/lib/x86_64-linux-gnu/libc.so.6(+0x29d90) [0x7a63c2629d90]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80) [0x7a63c2629e40]
/home/soumya/Downloads/rosetta.source.release-371/main/source/bin/rosettaDNA.default.linuxgccrelease(+0x35f5) [0x5ee95f1985f5]

[END_BACKTRACE]

[FILE]: src/apps/public/rosettaDNA/rosettaDNA.cc
[LINE]: 42
[START_MESSAGE]
[ ERROR ] UtilityExitException
ERROR: Unimplemented method called.



[END_MESSAGE]
[END_CRASH_REPORT]

How do I go about with this now?

**Comment by roccomoretti** [comment, marked as answer]
The relevant error message is `Option file open failed for: 'design.flags'`

When you specify `@design.flags` on the command line, you're telling Rosetta that there's a file named `design.flags` in the current directory which it should open and read the options from.

Generally speaking, when you run Rosetta you should `cd` into the "working directory" for the project which contains all your input file and where you want the output files to go. (There's some flexibility, in that you can specify relative paths, but the general concept is that you should be in your "base" directory for that particular project - the one where all your working files are relative to - before you launch Rosetta.)

**Comment by intomybioverse** [comment]
Thank you so much! It is working now. 

Also, can you please help me with generating a score file that gives the binding energy and other important parameters?

**Comment by roccomoretti** [comment]
Most Rosetta runs should automatically output a scorefile with the relevant data in it.

If you're looking for binding energy specifically, the typical application for protein-protein energy is the [InterfaceAnalyzer](https://docs.rosettacommons.org/docs/latest/application_documentation/analysis/interface-analyzer) -- I don't know how well that will work with DNA complexes, though. (It might - I've never tried it. You may want to adjust the energy function used, though.) A better bet might be to look through the related literature and see how the person who originally developed the protocol did the post-analysis.

**Comment by roccomoretti** [comment]
Marking this as "closed", as I hope your question has been answered. If that assessment is incorrect, feel free to re-open or open up a new discussion with more details.
